### PR TITLE
Latest code

### PR DESCRIPTION
**Pull request by nareshterli4**
latest code

### CODE REVIEW

File: swordfish/system_details.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            for k, v in data.items():
+                if k == "AllocatedBytes" and data['AllocatedBytes']:
+                    c['total'] = data['AllocatedBytes']
+                if k == "ConsumedBytes" and data['ConsumedBytes']:
+                    c['consumed'] = data['ConsumedBytes']
+
+            c['remaining'] = c['total'] - c['consumed']
+
+        except Exception as e:
+            return {"error": "Unable read file because of following error::{}".format(e)}, 500
```
Comment (rahlvers) [inline swordfish/system_details.py]: Fix: Change repeated instances of  "Unable read file" to "Unable to read file"

File: swordfish/constants.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        'data_security_los_capabilities': 'DataSecurityLoSCapabilities/',
+        'data_storage_los_capabilities': 'DataStorageLoSCapabilities/',
+        'ioconnectivity_los_capabilities': 'IOConnectivityLoSCapabilities/',
+        'ioperformance_los_capabilities': 'IOPerformanceLoSCapabilities/',
+        'endpoints': 'Endpoints/',
+	'storage_subsystems':'StorageSubsystems/',
+        'file_systems': 'FileSystems/'
+
+    },
+    'AddService':
```
Comment (rahlvers) [inline swordfish/constants.py]: Should there also be a modify and / or Delete here?

File: Resources/AddService/IPAddress.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{"Device": [{"IPAddress": "437XR1138R2"}, {"IPAddress": "437XR1138R3"}, {"IPAddress": "437XR1138R4"}, {"IPAddress": "183.25.16.12"}, {"url": "http://183.25.16.12:5000/redfish/v1/", "IPAddress": "183.25.16.12"}]}
```
Comment (rahlvers) [inline Resources/AddService/IPAddress.json]: What is the purpose of this file?  These seem to be specific to your configuration.

**Comment by rahlvers** [comment]
Need installation instructions - as noted in separate comments from Don:  
 * Target instructions for someone starting from scratch
 * Include python and any other base system installation requirements
  * Include where to get redfish emulator source
  * Include how to integrate swordfish source with redfish emulator

File: swordfish/system_details.py (hunk 1 above)
Comment (alansill) [inline swordfish/system_details.py]: ServiceDetails is misspelled as "ServiceDetais" in the class definition.

File: swordfish/system_details.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
+import json
+
+from flask import jsonify, request
+from flask.ext.restful import Resource
+import psutil
+
+from constants import PATHS
+
+
+class SystemDetaislAPI(Resource):
```
Comment (rahlvers) [inline swordfish/system_details.py]: Fix typo in class name:  SystemDetaislAPI should be SystemDetailsAPI

File: swordfish/constants.py (hunk 2 above)
Comment (nareshterli4) [inline swordfish/constants.py]: No, we wont write any modify/delete option in Conatants.py, we have to write in StorageServices.py if needed.

File: swordfish/system_details.py (hunk 4 above)
Comment (nareshterli4) [inline swordfish/system_details.py]: Fixed and raised PR

File: swordfish/system_details.py (hunk 1 above)
Comment (nareshterli4) [inline swordfish/system_details.py]: Fixed and raised PR

File: Resources/AddService/IPAddress.json (hunk 3 above)
Comment (nareshterli4) [inline Resources/AddService/IPAddress.json]: This file is used to maintain all the system configurations added as Add SwordFish Service through WEB -UI

Comment (ddeel) [inline Resources/AddService/IPAddress.json]: The Swordfish emulator should not need to include this file or any other special support for the WEB-UI. The WEB-UI needs to be able to operate without this kind of support from the emulator.

**Comment by ddeel** [comment]
The current Pull Request includes a "SNIA UI Functionality" document. Presumably, this document is here mainly as a way to facilitate talking about the emulator and what it must be able to do.

**Comment by ddeel** [comment]
The latest code for the Swordfish emulator (from this pull request's branch) gives this warning:
        ExtDeprecationWarning: Importing flask.ext.restful is deprecated, use flask_restful instead.
            from flask.ext.restful import Resource

This warning is because storage_services.py does not yet use flask_restful instead of flask.ext.restful. As of the last update, the Redfish emulator uses flask_restful, so it does not give this warning. Can this be fixed, to keep the Swordfish emulator more up-to-date with the Redfish emulator?

**Comment by rahlvers** [comment]
Closed - expecting new code drop 11/17/17.

**Pull request closed without merging**